src/objects/selection.ts

```diff
@@ -50,7 +50,23 @@ export function selectItem(item: GraphItemSelection | null): void {
 }
 
 export function isSelected(item: GraphItemSelection | SelectableItem | null): boolean {
-  return item?.id === selected?.id
+  if (item === null || selected === null) {
+    return false
+  }
+
+  if (Array.isArray(item) && 'ids' in selected) {
+    return item.length === selected.ids.length && selected.ids.every(id => item.includes(id))
+  }
+
+  if ('ids' in item && 'ids' in selected) {
+    return item.ids.length === selected.ids.length && selected.ids.every(id => item.ids.includes(id))
+  }
+
+  if ('id' in item && 'id' in selected) {
+    return item.id === selected.id
+  }
+
+  return false
 }
```

Comment:
Open to feedback on this. It's flexible, which is nice when using it, but may be getting hard to maintain. I anticipate future layers to have either an id, or `ids` like artifacts, so this hopefully wouldn't require any more adjustment as additional layers are introduced.

Reply:
I would probably expand on this just for readability and also type safety. I'd write separate `isSelectedNode`, `isSelectedArtifact` utilities and use those directly when possible. And then have `isSelected` call those individual utilities. 

Also I'd write type guards for the various selection kinds. So a type guard for `isNodeSelection`, `isArtifactSelection`, and `isArtifactsSelection` (or `isArtifactClusterSelection`) that way you can easily narrow a selection.